core-bundle/src/EventListener/MergeHttpHeadersListener.php

```diff
@@ -101,11 +102,24 @@ public function reset(): void
     /**
      * Fetches and stores HTTP headers from PHP.
      */
-    private function fetchHttpHeaders(): void
+    private function fetchHttpHeaders(Request $request): void
     {
         $headers = $this->headerStorage->all();
+        $session = $request->getSession();
+        $deprectatedHeaders = array_filter(
+            $headers,
+            static function ($header) use ($session): bool {
+                // Ignore Set-Cookie header set by PHP when using NativeSessionStorage
+                if (str_starts_with($header, "Set-Cookie: {$session->getName()}=")) {
```

Comment:
Should we check for `hasSession()` here?
```suggestion
        $session = $request->hasSession() ? $request->getSession() : null;
        $deprectatedHeaders = array_filter(
            $headers,
            static function ($header) use ($session): bool {
                // Ignore Set-Cookie header set by PHP when using NativeSessionStorage
                if ($session && str_starts_with($header, "Set-Cookie: {$session->getName()}=")) {
```

Reply:
Not sure. There is a lot of places where we execute `$request->getSession()` without checking for `hasSession()`. But it certainly won't hurt.